tests/phpunit/tests/taxonomy.php

```diff
@@ -125,7 +125,12 @@ public function test_taxonomy_exists_unknown() {
 		$this->assertFalse( taxonomy_exists( rand_str() ) );
 		$this->assertFalse( taxonomy_exists( '' ) );
 		$this->assertFalse( taxonomy_exists( 0 ) );
-		$this->assertFalse( taxonomy_exists( null ) );
+		$this->assertFalse( taxonomy_exists( 1 ), 'Did not return false with (int) 1' );
+		$this->assertFalse( taxonomy_exists( 0.0 ), 'Did not return false with (float) 0.0' );
+		$this->assertFalse( taxonomy_exists( 1.1 ), 'Did not return false with (float) 1.1' );
+		$this->assertFalse( taxonomy_exists( null ), 'Did not return false with null' );
+		$this->assertFalse( taxonomy_exists( array(), 'Did not return false with an array' ) );
+		$this->assertFalse( taxonomy_exists( new stdClass(), 'Did not return false with an object' ) );
```

Comment:
These tests should ideally be in a data provider. Given the existing tests aren't, this seems like a matter for another ticket.